components/scream/src/physics/shoc/shoc_calc_shoc_vertflux_impl.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef SHOC_CALC_SHOC_VERTFLUX_IMPL_HPP
+#define SHOC_CALC_SHOC_VERTFLUX_IMPL_HPP
+
+#include "shoc_functions.hpp" // for ETI only but harmless for GPU
+
+namespace scream {
+namespace shoc {
+
+template<typename S, typename D>
```

Comment:
This is called in a kernel, so you'll need a KOKKOS_FUNCTION annotation.

Reply:
Thanks, I hadn't tested this on GPU.